src/averageAges.js

```diff
@@ -9,17 +9,25 @@
  * To calculate century:
  * Divide year of person's death by 100: Math.ceil(person.died / 100)
  *
+ *
+ *
+ *
  * @param {object[]} people
  * @param {number} century - optional
  *
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const mail = (century)
+    ? people
+      .filter(person => person.sex === 'm'
+        && Math.ceil(person.died / 100) === century)
+    : people
+      .filter(person => person.sex === 'm');
+
+  return mail
+    .map(person => person.died - person.born)
+    .reduce((a, b) => a + b) / mail.length;
 }
```

Comment:
could you move to the separate function & reuse it?
also, what is a & what is b?)
```suggestion
  return getAverageAge(men)
```